tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -2618,6 +2652,10 @@ public static void initAndStartAppMaster(final DAGAppMaster appMaster,
       final Configuration conf) throws IOException,
       InterruptedException {
 
+    if (appMaster.tezThreadDumpHelper != null) {
+      appMaster.tezThreadDumpHelper.schedulePeriodicThreadDumpService("AppMaster");
```

Comment:
if you start this in the scope of a DAG, you can easily pick the actual dagName here instead of "AppMaster"

Reply:
Yep, got the dag name in the new logic